simplex/engine.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
```

Comment:
is it possible to strip out the engine code addition from this PR? I think the chain router and protobuf is enough to capture some functionality, while I'm not sure the engine changes provide significant functionality to justify being part of this PR.